content/spaces/locking.textile

```diff
@@ -0,0 +1,301 @@
+
+A member must have been "entered":/spaces/space#enter into the space to acquire a lock.
+
+The following is an example of attempting to acquire a lock:
+
+```[javascript]
+const acquireLock = await space.locks.acquire(id);
+```
+
+The following is an example of passing a set of @attributes@ when trying to acquire a lock:
+
+```[javascript]
+const lockAttributes = new Map();
+lockAttributes.set('component', 'cell-d3');
+const acquireLock = await space.locks.acquire(id, { lockAttributes });
+```
+
+The following is an example payload returned by @space.locks.acquire()@. The promise will resolve to a lock request with the @pending@ status:
+
+```[json]
+{
+  "id": "s2-d14",
+  "status": "pending",
+  "timestamp": 1247525689781,
+  "attributes": {
+    "componentId": "cell-d14"
+  }
+}
+```
+
+Once a member requests a lock by calling @acquire()@, the lock is temporarily in the "pending state":#states. An event will be emitted based on whether the lock request was successful (a status of @locked@) or invalidated (a status of @unlocked@). This can be "subscribed":#subscribe to in order for the client to know whether their lock request was successful or not.
+
+h2(#release). Release a lock
+
+Use the @release()@ method to explicitly release a lock once a member has finished editing the related component. For example, the @release()@ method can be called once a user clicks outside of the component, such as clicking on another cell within a spreadsheet. Any UI indications that the previous cell was locked can then be cleared.
+
+The following is an example of releasing a lock:
+
+```[javascript]
+await space.locks.release(id);
+```
+
+Releasing a lock will emit a lock event with a "lock status":#states of @unlocked@.
+
+<aside data-type='note'>
+<p>When a member "leaves":/spaces/space#leave a space, their locks are automatically released.</p>
+</aside>
+
+h2(#subscribe). Subscribe to lock events
+
+Subscribe to lock events by registering a listener. Lock events are emitted whenever the "lock state":#states transitions into @locked@ or @unlocked@. Use the @subscribe()@ method on the @locks@ namespace of the space to receive updates.
+
+All lock events are @update@ events. When a lock event is received, UI components can be updated to add and remove visual indications of which member is locking them, as well as enabling and disabling the ability for other members to edit them.
+
+The following is an example of subscribing to lock events:
+
+```[javascript]
+space.locks.subscribe('update', (lock) => {
+  console.log(lock);
+});
+```
+
+The following is an example payload of a lock event:
+
+```json
+{
+  "member": {
+    "clientId": "smango",
+    "connectionId": "hs343gjsdc",
+    "isConnected": true,
+    "profileData": {
+      "username": "Saiorse Mango"
+    },
+    "location": {
+      "slide": "sheet-2",
+      "component": "d-14"
+    },
+    "lastEvent": {
+      "name": "update",
+      "timestamp": 1247525689781
+    }
+  },
+  "request": {
```

Comment:
This has now been flattened, so there is no `request` field, all the fields within `request` here are now at the top-level, see https://github.com/ably-labs/spaces/pull/151.

Reply:
Thanks, I missed that one. Updated that payload, the table and `getAll()` payload in [0f7e2bd](https://github.com/ably/docs/pull/1967/commits/0f7e2bdebb72e8a3f291c97deca282f36af69782)

Reply:
👍